codegen/main.go

```diff
@@ -0,0 +1,461 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"cmp"
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"slices"
+	"strings"
+	"text/template"
+	"time"
+	"unicode"
+
+	"golang.org/x/exp/maps"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	outF  = flag.String("outF", "", "Output file")
```

Comment:
NIT: Let's make the variable name more descriptive - outFile, inFile, templateFile since these are exposed flags